packages/eslint-plugin/tests/rules/prefer-nullish-coalescing.test.ts

```diff
@@ -30,31 +30,42 @@ const nullishTypes = ['null', 'undefined', 'null | undefined'];
 const ignorablePrimitiveTypes = ['string', 'number', 'boolean', 'bigint'];
 
 function typeValidTest(
-  cb: (type: string) => ValidTestCase<Options> | string,
+  cb: (type: string, equals: '' | '=') => ValidTestCase<Options> | string,
```

Comment:
[Testing] We've been moving away from auto-generated tests. They tend to generate a huge spread of unit tests that each test ever so slightly different variations of the same thing. For example, this one now means each intricacy passed to `typeValidTest` is tested under both values of `equals`.

IMO it'd be better to instead have specific targeted tests for each behavior under test.

Change request: revert any additions to these test generation functions, and instead write new tests manually.